mmv1/third_party/terraform/data_sources/data_source_alloydb_supported_database_flags.go.erb

```diff
@@ -0,0 +1,242 @@
+							Computed:    true,
+							Optional:    true,
+							Description: ``,
+							MaxItems:    1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"allowed_values": {
+										Type:        schema.TypeList,
+										Computed:    true,
+										Optional:    true,
+										Description: `The list of allowed values, if bounded. This field will be empty if there is a unbounded number of allowed values.`,
+										Elem:        &schema.Schema{Type: schema.TypeString},
+									},
+								},
+							},
+						},
+						"integer_restrictions": {
+							Type:        schema.TypeList,
+							Computed:    true,
+							Optional:    true,
+							Description: ``,
+							MaxItems:    1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"min_value": {
+										Type:        schema.TypeInt,
+										Computed:    true,
+										Optional:    true,
+										Description: `The minimum value that can be specified, if applicable.`,
+									},
+									"max_value": {
+										Type:        schema.TypeInt,
+										Computed:    true,
+										Optional:    true,
+										Description: `The maximum value that can be specified, if applicable.`,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func dataSourceAlloydbSupportedDatabaseFlagsRead(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+	userAgent, err := generateUserAgentString(d, config.UserAgent)
+	if err != nil {
+		return err
+	}
+	project, err := getProject(d, config)
+	if err != nil {
+		return fmt.Errorf("Error fetching project: %s", err)
+	}
+	billingProject := project
+	if bp, err := getBillingProject(d, config); err == nil {
+		billingProject = bp
+	}
+	location := ""
+	if v, ok := d.GetOk("location"); ok {
+		location = v.(string)
+	}
+	if location == "" {
+		return fmt.Errorf("Location cannot be empty")
+	}
+	var supportedDatabaseFlagIterator *alloydb.SupportedDatabaseFlagIterator
+	locsReq := new(alloydbpb.ListSupportedDatabaseFlagsRequest)
+	locationsClient := config.NewAlloydbClient(userAgent)
+	if locationsClient == nil {
+		return fmt.Errorf("Failed to call the API to fetch the supported database flags")
+	}
+	err = nil
+	err = retryTime(func() error {
+<% if version == "ga" -%>
+		url := fmt.Sprintf("v1/projects/%s/locations/%s/supportedDatabaseFlags", billingProject, location)
+<% else -%>
+		url := fmt.Sprintf("v1beta/projects/%s/locations/%s/supportedDatabaseFlags", billingProject, location)
+<% end -%>
+		supportedDatabaseFlagIterator = locationsClient.ListSupportedDatabaseFlags(config.context, locsReq, gax.WithPath(url))
+		return nil
+	}, 5)
+	if err != nil {
+		return handleNotFoundError(err, d, fmt.Sprintf("Supported Database flags %q", d.Id()))
+	}
+
+	var supportedDatabaseFlags []map[string]interface{}
+	for {
+		supportedDatabaseFlag := make(map[string]interface{})
+		flag, err := supportedDatabaseFlagIterator.Next()
```

Comment:
Does the `ListSupportedDatabaseFlags` call return a slice of supported database flags? Can we just use a for loop over the slice without using an iterator & while loop?

Reply:
The alloydb go client returns an iterator instead of the response directly. This does adds an added advantage of iterator calling the next batch of database flags by passing next page token automatically.